proxmox/datadog_checks/proxmox/check.py

```diff
@@ -15,33 +23,67 @@ class ProxmoxCheck(AgentCheck, ConfigMixin):
 
-    def check(self, _):
-        try:
-            response = self.http.get(f"{self.config.proxmox_server}/version")
-            response.raise_for_status()
-            response_json = response.json()
-            version = response_json.get("data", {}).get("version")
-            self.set_metadata('version', version)
-            self.gauge("api.up", 1, tags=self.base_tags)
+    def _submit_resource_metrics(self, resource, tags, hostname):
+        for metric_name, metric_name_remapped in RESOURCE_METRIC_NAME.items():
+            metric_value = resource.get(metric_name)
+            metric_method = self.count if metric_name in RESOURCE_COUNT_METRICS else self.gauge
+            if metric_value is not None:
+                metric_method(f'{metric_name_remapped}', metric_value, tags=tags, hostname=hostname)
 
+    def _get_vm_hostname(self, vm_id, vm_name, node):
+        try:
+            url = f"{self.config.proxmox_server}/nodes/{node}/qemu/{vm_id}/agent/get-host-name"
+            hostname_response = self.http.get(url)
+            hostname_json = hostname_response.json()
         except (HTTPError, InvalidURL, ConnectionError, Timeout, JSONDecodeError) as e:
-            self.log.error(
-                "Encountered an Exception when hitting the Proxmox API %s: %s", self.config.proxmox_server, e
+            self.log.info(
+                "Failed to get hostname for vm %s on node %s; endpoint: %s; %s",
+                vm_id,
+                node,
+                self.config.proxmox_server,
+                e,
             )
-            self.gauge("api.up", 0, tags=self.base_tags)
-            return
+            hostname_json = {}
+        hostname = hostname_json.get("data", {}).get("result", {}).get("host-name", vm_name)
+        return hostname
+
+    def _collect_performance_metrics(self):
+        metrics_response = self.http.get(f"{self.config.proxmox_server}/cluster/metrics/export")
+        metrics_response_json = metrics_response.json()
+        metrics = metrics_response_json.get('data', {}).get('data', [])
+
+        for metric in metrics:
+            resource_id = metric.get('id')
+            resource = self.all_resources.get(resource_id, {})
+            metric_value = metric.get('value')
+            metric_name = metric.get('metric')
+            metric_type = metric.get('type')
+            metric_name_remapped = PERF_METRIC_NAME.get(metric_name)
+            hostname = resource.get('hostname')
+            tags = resource.get('tags', [])
+            if not resource or metric_name_remapped is None:
+                self.log.debug(
+                    "Invalid metric entry found; metric name: %s, resource id: %s", metric_name_remapped, resource_id
+                )
+                continue
 
+            metric_method = self.count if metric_type == 'derive' else self.gauge
+            metric_method(metric_name_remapped, metric_value, tags=tags, hostname=hostname)
```

Comment:
Will self.tags be attached here in these metrics?

Reply:
Yes- they're set here: https://github.com/DataDog/integrations-core/pull/20631/files/4c7c5fbbf848ab5d9bef2c6bcc262c377700145b#diff-6d85066f4a20fb3eb1c6be9644479d3b217ce300f5e4c71fd03d04708de15af9R138